libs/vkd3d/vkd3d_private.h

```diff
@@ -638,13 +639,15 @@ struct vkd3d_allocate_memory_info
     uint32_t flags;
     VkBufferUsageFlags explicit_global_buffer_usage;
     VkMemoryPropertyFlags optional_memory_properties;
+    float vk_memory_priority;
 };
 
 struct vkd3d_allocate_heap_memory_info
 {
     D3D12_HEAP_DESC heap_desc;
     void *host_ptr;
     uint32_t extra_allocation_flags;
+    float vk_memory_priority; // a heap allocation can actually be a dedicated allocation under the hood (xxx: I think that's true?), in which case we can still fulfil a memory priority
```

Comment:
C89 style comments.

Reply:
And yes, a ID3D12Heap can map 1:1 to a VkDeviceMemory, but only if we don't need suballocation.